addons/web_editor/controllers/main.py

```diff
@@ -888,7 +887,7 @@ def save_library_media(self, media):
                 attachment = IrAttachment.with_user(SUPERUSER_ID).create(attachment_data)
             if media[id]['is_dynamic_svg']:
                 colorParams = werkzeug.urls.url_encode(media[id]['dynamic_colors'])
-                attachment['url'] = '/web_editor/shape/illustration/%s?%s' % (slug(attachment), colorParams)
+                attachment['url'] = '/web_editor/shape/illustration/%s?%s' % (request.env['ir.http']._slug(attachment), colorParams)
```

Comment:
I guess this was kinda the same before this PR but web_editor does not depend on http_routing so technically, this is wrong. Not sure if this is more wrong than before though? Maybe yes: python utils can be accessed even though the app is not installed maybe? :shrug: 

Maybe the `_slug` method and everything could be defined in the base ir.http model? :shrug:

Reply:
After this commit, since inheritance is managed by odoo depending on the installed modules, the method does not return the same thing.
For example, if we only use base module, the urls do not use the name of the elements, only their id (like when the url is parsed). This behavior changes when website is installed. This is indeed an inheritance and not a behavior of the base module.

Reply:
I understood that but that was not the point of my comment :wink: 
The point is: web_editor uses something which comes from http_routing, so this is technically wrong...? Or I am missing something.

Reply:
Yes it is wrong to use it since it does not depend on it. This must be corrected so that the behavior is correct even if the other module is not installed.